classes/kohana/encrypt.php

```diff
@@ -1,213 +1,121 @@
 <?php defined('SYSPATH') or die('No direct script access.');
 /**
  * The Encrypt library provides two-way encryption of text and binary strings
- * using the [Mcrypt](http://php.net/mcrypt) extension, which consists of three
- * parts: the key, the cipher, and the mode.
- *
- * The Key
- * :  A secret passphrase that is used for encoding and decoding
- *
- * The Cipher
- * :  A [cipher](http://php.net/mcrypt.ciphers) determines how the encryption
- *    is mathematically calculated. By default, the "rijndael-128" cipher
- *    is used. This is commonly known as "AES-128" and is an industry standard.
- *
- * The Mode
- * :  The [mode](http://php.net/mcrypt.constants) determines how the encrypted
- *    data is written in binary form. By default, the "nofb" mode is used,
- *    which produces short output with high entropy.
- *
+ * using the [Sodium](https://www.php.net/manual/en/book.sodium.php) extension.
+
  * @package    Kohana
  * @category   Security
  * @author     Kohana Team
  * @copyright  (c) 2007-2012 Kohana Team
  * @license    http://kohanaframework.org/license
  */
 class Kohana_Encrypt {
-
 	/**
-	 * @var  string  default instance name
+	 * @var string
 	 */
-	public static $default = 'default';
+	private static $default = 'default';
 
 	/**
-	 * @var  array  Encrypt class instances
+	 * @var array
 	 */
-	public static $instances = array();
+	private static $instances = [];
 
 	/**
-	 * @var  string  OS-dependent RAND type to use
+	 * @var string
 	 */
-	protected static $_rand;
+	private $key;
 
 	/**
-	 * Returns a singleton instance of Encrypt. An encryption key must be
-	 * provided in your "encrypt" configuration file.
-	 *
-	 *     $encrypt = Encrypt::instance();
-	 *
-	 * @param   string  $name   configuration group name
-	 * @return  Encrypt
+	 * @param string $key Encryption key (hexadecimal)
 	 */
-	public static function instance($name = NULL)
+	public function __construct($key)
 	{
-		if ($name === NULL)
-		{
-			// Use the default instance name
-			$name = Encrypt::$default;
-		}
-
-		if ( ! isset(Encrypt::$instances[$name]))
-		{
-			// Load the configuration data
-			$config = Kohana::$config->load('encrypt')->$name;
-
-			if ( ! isset($config['key']))
-			{
-				// No default encryption key is provided!
-				throw new Kohana_Exception('No encryption key is defined in the encryption configuration group: :group',
-					array(':group' => $name));
-			}
-
-			if ( ! isset($config['mode']))
-			{
-				// Add the default mode
-				$config['mode'] = MCRYPT_MODE_NOFB;
-			}
-
-			if ( ! isset($config['cipher']))
-			{
-				// Add the default cipher
-				$config['cipher'] = MCRYPT_RIJNDAEL_128;
-			}
-
-			// Create a new instance
-			Encrypt::$instances[$name] = new Encrypt($config['key'], $config['mode'], $config['cipher']);
-		}
-
-		return Encrypt::$instances[$name];
+		$this->key = hex2bin($key);
```

Comment:
I'd add a length check and throw an exception if the key is not the correct length to prevent and configuration mistakes.